jaeger_client/codecs.py

```diff
@@ -295,15 +295,25 @@ def inject(self, span_context, carrier):
         elif span_context.flags & SAMPLED_FLAG == SAMPLED_FLAG:
             carrier[self.sampled_header] = '1'
 
+    @staticmethod
+    def _get_header_value_hex(carrier, header_lc_to_header_orig, header_lc):
+        header = header_lc_to_header_orig.get(header_lc)
+        if header is None:
+            return None
+        header_value = carrier.get(header)
+        if header_value is not None:
+            header_value = header_to_hex(header_value)
+        return header_value
+
     def extract(self, carrier):
         if not isinstance(carrier, dict):
             raise InvalidCarrierException('carrier not a dictionary')
         lowercase_keys = dict([(k.lower(), k) for k in carrier])
```

Comment:
I wonder if it would be simpler and more efficient to just loop through the carrier keys, e.g.
```
for key, value in six.itemitems(carrier):
    if value is None:
        continue  # is this even possible in Python?
    lowerKey = key.lower()
    if lowerKey == self._trace_header_lc:
        # parse
    elif lowerKey == self._span_header_lc:
        # parse
```
This avoids creating an intermediate dictionary, and partially addresses your issue